controllers/lookup/druid_client.go

```diff
@@ -0,0 +1,163 @@
+package lookup
+
+import (
+	"encoding/json"
+	"fmt"
+	internalhttp "github.com/datainfrahq/druid-operator/pkg/http"
+	"k8s.io/apimachinery/pkg/types"
+	"net/http"
+	"reflect"
+	"time"
+)
+
+type DruidClient struct {
```

Comment:
Non-blocking - I think that what you did here might be a good interface for all reconcilers we might create that works with the Druid API. 
Do you think we can have this as an interface and maybe create an abstraction for other reconcilers to use? 
You don't have to implement it here, or at all :) would appreciate your opinion

Reply:
Thank you, happy to you like it! My goal with this struct were to collect the parts that interact with the Druid API into one place. I can definitely see how that API interaction being a common problem for all reconcilers working with Druid state.

> Do you think we can have …

I see two ways of interpret this question. If you're asking for my permission to build on top of the ideas within the code, 1. how kind of you to ask(!), 2. please do! If you're asking if it would be feasible to build an abstraction on top of it, then here's a bit too long ramble of how one could do that!

Most of the methods of this type (e.g. `delete`, or `upsert`) are pretty low-level in the sense that they are just wrappers around the HTTP API. I think having one type/package that exposes this functionality as methods/functions to abstract away the HTTPness of the API would probably be helpful.

Some of the other methods that `DruidClient` expose, such as `Reconcile` or `GetStatus`, are a lot more logic oriented. Perhaps there are common patterns across the different reconcilers that might be worth abstracted cleanly. Perhaps the reconciliation logic is one such place. Given some interface that exposes functionality can list current instances of a resources, as well as create, replace and delete instance of that resource, one could probably create a function that consumes such an interface as well as a list of desired instances of that resource. There of course is the problem of slight variations in different applications of that pattern. The lookup resource has an example of such a slight variation of the base pattern, namely: the resource space does not start initialised, and does not support retrieving or listing instances before it is initialised. Of course, that specific example is easily solved by initialising the lookup space before handing it of to the hypothetical generic reconciliation function. However, that complication was likely easy to fix with in that abstraction since I drafted it within the context of lookup reconciliation. While I still lack the much wider context of the other reconcilers in the operator, I would be more than happy to contribute to the discussion!